src/edgekv/ekv-helper.ts

```diff
@@ -0,0 +1,339 @@
+        let tokenData = tokenContent[key];
+        let nameSpaceContent = `\n"${key}" : { \n"name": "${tokenData["name"]}",\n"value" : "${tokenData["value"]}"\n}`;
+        token.push(nameSpaceContent);
+    });
+    return tkn_var + token.toString() + tkn_export;
+}
+
+/**
+ * If only directory is specified without tgz, we create new token file and place it in the save path.
+ * If token file already exists, token will be updated. Users can place this token file when they place it in th bundle
+ * @param savePath 
+ * @param overWrite 
+ * @param createdToken 
+ * @param decodedToken 
+ * @param nameSpaceList 
+ */
+export function createTokenFileWithoutBundle(savePath, overWrite, createdToken, decodedToken, nameSpaceList) {
+
+    let msg = `Token in ${savePath}/edgekv_tokens.js was successfully updated.`;
+
+    // if token file does not exist, create new file
+    let tokenFilePath = savePath + "/edgekv_tokens.js";
+    let tokenFileContent = "";
+    let tokenContent = [];
+    if (!checkIfFileExists(tokenFilePath)) {
+        for (let ns of nameSpaceList) {
+            tokenContent[ns] = { "name": createdToken["name"], "value": createdToken["value"] };
+        }
+    }
+    // update existing token file
+    else {
+        var text = fs.readFileSync(tokenFilePath, 'utf8');
+        // get existing token content from file
+        let existingTokenContent = validateAndGetExistingTokenContent(text, createdToken, decodedToken, nameSpaceList);
+        // updated content
+        tokenContent = updateTokenContent(existingTokenContent, nameSpaceList, createdToken, decodedToken, overWrite);
+    }
+    tokenFileContent = constructTokenFile(tokenContent);
+    fs.writeFile(tokenFilePath, tokenFileContent, function (err, file) {
+        if (err) {
+            cliUtils.logWithBorder(`ERROR: Unable to create a token file. Add the token in file edgekv_tokens.js and place it in your tar bundle`);
+            response.logToken(createdToken["name"], createdToken["value"], decodedToken, nameSpaceList, false);
+            process.exit(1);
+        }
+    });
+    cliUtils.logWithBorder(msg);
+    response.logToken(createdToken["name"], createdToken["value"], decodedToken, nameSpaceList, true);
+}
+
+/**
+ * Validates the static content of the token 
+ * If valid parses the content and returns it
+ * @param data 
+ * @param createdToken 
+ * @param decodedToken 
+ * @param nameSpaceList 
+ */
+function validateAndGetExistingTokenContent(data, createdToken, decodedToken, nameSpaceList) {
+    let tokenList = data.split("=");
+    let tokenContent = [];
+
+    if (tokenList.length == 0 || tokenList[0].indexOf("var edgekv_access_tokens") == -1) {
+        cliUtils.logWithBorder("ERROR : Not a valid EdgeKV Access Token file (missing 'edgekv_access_tokens' var assignment)!");
+        response.logToken(createdToken["name"], createdToken["value"], decodedToken, nameSpaceList, false);
+        process.exit(1);
```

Comment:
Is process.exit the right way to go in this file vs. using  cliUtils.logAndExit ?

Reply:
right! will update cliUtils.logAndExit ?